MqttOutput.go

```diff
@@ -0,0 +1,70 @@
+package dnstap
+
+import (
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"google.golang.org/protobuf/proto"
+)
+
+// MqttOutput implements a dnstap Output that publishes dnstap data to an MQTT broker.
+type MqttOutput struct {
+	outputChannel chan []byte
+	client        mqtt.Client
+	baseTopic     string
+	qos           byte
+	wait          chan bool
+	log           Logger
+}
+
+// NewMqttOutput creates a MqttOutput for publishing dnstap data to an MQTT broker.
+func NewMqttOutput(opts *mqtt.ClientOptions, baseTopic string, qos byte) (o *MqttOutput, err error) {
```

Comment:
Since we are not using defer statements that must inspect return values, we do not need to name them. This simplifies the code below.

```suggestion
func NewMqttOutput(opts *mqtt.ClientOptions, baseTopic string, qos byte) (*MqttOutput, error) {
```